src/Symfony/Bundle/FrameworkBundle/Console/Descriptor/TextDescriptor.php

```diff
@@ -374,6 +374,35 @@ protected function describeContainerDefinition(Definition $definition, array $op
         $options['output']->table($tableHeaders, $tableRows);
     }
 
+    /**
+     * {@inheritdoc}
+     */
+    protected function describeContainerDeprecations(ContainerBuilder $builder, array $options = []): void
+    {
+        $containerDeprecationFilePath = sprintf('%s/%sDeprecations.log', $builder->getParameter('kernel.cache_dir'), $builder->getParameter('kernel.container_class'));
+        if (!file_exists($containerDeprecationFilePath)) {
+            $options['output']->warning('The deprecation file does not exist, please try warming the cache first.');
+
+            return;
+        }
+
+        $logs = unserialize(file_get_contents($containerDeprecationFilePath));
+        if (0 === \count($logs)) {
+            $options['output']->success('There are no deprecations in the logs!');
+
+            return;
+        }
+
+        $formattedLogs = [];
+        $remainingCount = 0;
+        foreach ($logs as $log) {
+            $formattedLogs[] = sprintf("%sx: %s \n      in %s:%s", $log['count'], $log['message'], $log['file'], $log['line']);
```

Comment:
we should use `<href=%s>%s:%d</>` here, to generate IDE links (which means using the FileLinkFormatter service to compute it)